mautic/mautic
# Remove dwc session, add cache

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? | N
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description: 
Dynamic content cannot use session for tracking because of session cookie does not have samesite none.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Set up dynamic content - https://www.mautic.org/docs/en/dwc/index.html. You need a campaign that has Campaign form as a source, then Request dynamic content, some condition and Push dynamic content. In the page you need the form code, tracking code and the code below:
```
<div data-slot="dwc" data-param-slot-name="[yourSlotName]">
  <h1>Dynamic web content for myslot</h1>
</div>
```
2. When you use it, you see that set-cookie header is there without samesite=none. It is on the file /dwc/[slotName]. The set-cookie is there when you submit the form a reload page.

#### Steps to test this PR:
1. Load up this PR and rebase staging (there are some other sessions fixes)
2. Set up dynamic content - https://www.mautic.org/docs/en/dwc/index.html
3. Check that all cookies have samesite-none. And dynamic content works well.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
